deps/build.jl

```diff
@@ -1,7 +1,7 @@
-# If BLAS was compiled with MKL, we want to use MKL for FFTs as well. Thus
-# we have to do this little dance to get around having to use BinDeps for
-# a library that's already linked to Julia.
-if Base.BLAS.vendor() === :mkl
+# If BLAS was compiled with MKL and the user wants MKL-based FFTs, we'll oblige.
+# In that case, we have to do this little dance to get around having to use BinDeps
+# for a library that's already linked to Julia.
+if get(ENV, "JULIA_FFTW_PROVIDER") == "MKL" && Base.BLAS.vendor() === :mkl
```

Comment:
The problem with this is that if you do a `Pkg.update()` it will "forget" your previous setting unless the environment variable is still set.  That's why in PyCall I save the setting in a file.

Reply:
Seems like the kind of thing that could go in a .juliarc.jl file

Reply:
Requiring users to not only set an env var, but to set it "permanently" in a file somewhere, seems awfully error prone.

Besides, environment variables live in a global namespace, e.g. they "leak" into Julia subprocesses, so I greatly prefer setups where you only need to set them temporarily.